File: howl-application/src/main/java/io/quartic/howl/storage/StorageBackend.java
package io.quartic.howl.storage;

import java.io.IOException;
import java.io.InputStream;
import java.util.Optional;

public interface StorageBackend {
    Optional<InputStreamWithContentType> get(String namespace, String objectName, Long version) throws IOException;
    Long put(String contentType, String namespace, String objectName, InputStream inputStream) throws IOException;
}

File: howl-application/src/main/kotlin/io/quartic/howl/storage/StorageBackend.kt
package io.quartic.howl.storage

import java.io.IOException
import java.io.InputStream

interface StorageBackend {
    @Throws(IOException::class)
    fun getData(coords: StorageCoords, version: Long?): InputStreamWithContentType?

    @Throws(IOException::class)
    fun putData(coords: StorageCoords, contentType: String?, inputStream: InputStream): Long?
}
